### ISSUE TITLE
2.4.12 Focus Not Obscured (Enhanced) (Level AAA)

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#553]
#### [2.4.12 Focus Not Obscured (Enhanced)](https://www.w3.org/TR/WCAG22/#focus-not-obscured-enhanced)

(Level AAA)

>When a [user interface component](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components) receives keyboard focus, no part of the component is hidden by author-created content.

##### Applying SC 2.4.12 Focus Not Obscured (Enhanced) to Non-Web Documents and Software
This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.12](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html#intent).

***
**For WCAG2ICT consideration:** Any notes needed for non-web context?  Any considerations for closed functionality?
***

**Comment by pday1** [comment]
Related SCs:
- 2.4.11 Focus not Obscured
- 2.1.2 No Keyboard Trap
- 2.4.7 Focus Visible

Applying SC 2.4.11 Focus not Obscured to non-web documents and non-web software
> This applies directly as written, and as described in Intent from Understanding Success Criterion 2.4.11.


Applying SC 2.1.2 No Keyboard Trap to non-web documents and non-web software

> This applies directly as written, and as described in Intent from Understanding Success Criterion 2.1.2, replacing “page” with “non-web document or non-web software” and “on the web page” with "in the non-web document or non-web software"; removing “See Conformance Requirement 5: Non-Interference”; and adjusting Note 1 to avoid the use of the normative term "must".
> 
> With these substitutions, it would read:
> 
> 2.1.2 No Keyboard Trap: If keyboard focus can be moved to a component of the [non-web document or non-web software] using a keyboard interface, then focus can be moved away from that component using only a keyboard interface, and, if it requires more than unmodified arrow or tab keys or other standard exit methods, the user is advised of the method for moving focus away.
> 
> NOTE 1
> Since any content that does not meet this success criterion can interfere with a user's ability to use the whole [non-web document or non-web software], [it would be necessary for] all content [in the non-web document or non-web software] (whether it is used to meet other success criteria or not) [to] meet this success criterion.
> 
> NOTE 2 (ADDED)
> Standard exit methods may vary by platform. For example, on many desktop platforms, the Escape key is a standard method for exiting.
> 
> NOTE 3 (ADDED) (FOR NON-WEB SOFTWARE)
> This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.
> 
> NOTE 4 (ADDED) (FOR NON-WEB SOFTWARE)
> See also the Comments on Closed Functionality.

SC problematic for closed:

> 2.1.2 No Keyboard Trap — This criterion applies when focus can be moved using a keyboard interface. In some ICT with closed functionality, tactile input like numeric keypads or other functional groups of keys may be available, but there is no mechanism for onscreen focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.



Applying SC 2.4.7 Focus Visible to non-web documents and non-web software

> This applies directly as written, and as described in Intent from Understanding Success Criterion 2.4.7.
> 
> NOTE (ADDED) (FOR NON-WEB SOFTWARE)
> See also the Comments on Closed Functionality.
> 

SC problematic for closed:

> 2.4.7 Focus Visible — Presumes that there is a mode of operation where focus can be moved and controlled by keyboard. Some ICT with closed functionality may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any mechanism for conveying focus because the user interface is designed not to need that. For example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, thus there is no need for a visible indicator and this success criterion would be satisfied.

**Comment by pday1** [comment]
Proposed content for review by TF is as follows.

##### Applying SC 2.4.12 Focus Not Obscured (Enhanced) to Non-Web Documents and Software

> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.12](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html#intent).

> NOTE (ADDED) (FOR NON-WEB SOFTWARE)
> This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.

> NOTE (ADDED) (FOR NON-WEB SOFTWARE)
> See also the Comments on Closed Functionality.

And proposed content for SC problematic for closed (derived from 2.4.7):

> 2.4.12 Focus Not Obscured (Enhanced)  — Presumes that there is a mode of operation where focus can be moved and controlled by keyboard. Some ICT with closed functionality may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any mechanism for conveying focus because the user interface is designed not to need that. For example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, thus there is no need for a visible indicator and this success criterion would be satisfied.

**Comment by daniel-montalvo** [comment]
I'd add "handling" or "management" after focus.

"yet not offer any mechanism for focus; " -> "yet not offer any mechanism for focus handling/management;"

**Comment by pday1** [comment]
Agreed text as per TF meeting of 7 May. https://www.w3.org/2026/05/07-wcag2ict-minutes.html#31a6


Proposed content for review by TF is as follows.

##### Applying SC 2.4.12 Focus Not Obscured (Enhanced) to non-web documents and non-web software

> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.12](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html#intent).

> NOTE (ADDED) (FOR NON-WEB SOFTWARE)
> This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus handling or management; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and this success criterion would be satisfied.

> NOTE (ADDED) (FOR NON-WEB SOFTWARE)
> See also the Comments on Closed Functionality.

And proposed content for SC problematic for closed (derived from 2.4.7):

> 2.4.12 Focus Not Obscured (Enhanced)  — Presumes that there is a mode of operation where focus can be moved and controlled by keyboard. Some ICT with closed functionality may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any mechanism for conveying focus because the user interface is designed not to need that. For example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, thus there is no need for a visible indicator and this success criterion would be satisfied.

**Comment by pday1** [comment]
In PR https://github.com/w3c/wcag2ict/pull/921 and now merged into main
